### ISSUE TITLE
Option to disable direct share

### ISSUE THREAD

**Issue by ourslfs** [Aliucord/Aliucord#413]
### What is it that you'd like to see?

After contacting someone, you get a list a discord contacts in your direct share menu. An option to disable such behavior would be nice(afaik you can disable Chooser Target Service Compat service with root, but you need root for that).

### Request Agreement

- [X] I did indeed check to make sure my feature request is original!

**Comment by rushiiMachine** [comment]
This isn't something thats on by default

**Comment by rushiiMachine** [comment]
oh never mind i thought you were talking about proximity contacts

**Comment by zt64** [comment]
I think this may require manifest modification? Maybe this should be transferred to the manager repository instead?

**Comment by rushiiMachine** [comment]
Yeah it definitely does. Problem is, adding patches like these on a case-by-case basis for a list of hardcoded ones is debatable.
The other suggestion i remember was PIP which would require additions to the injector as well as manifest patch.
Not sure what to do and how stuff like this should be handled properly

**Comment by zt64** [comment]
Okay so after some looking it might be doable without modifying manifest. In AppActivity there's a call to updateDirectShareTargets which could be patched. Might have to also unregister any existing share targets
